plugins/outputs/file/file.go

```diff
@@ -18,19 +19,47 @@ import (
 //go:embed sample.conf
 var sampleConfig string
 
+var ValidCompressionAlgorithmLevels = map[string][]int{
+	"zstd":     {1, 3, 7, 11},
+	"gzip":     {-2, -1, 1, 9},
+	"zlib":     {-2, -1, 1, 9},
+	"identity": {0},
+}
```

Comment:
Isn't this something that should be defined in the `internal` encoder package? How about unifying these across Telegraf (e.g. `low`, `standard`, `high`,  `extreme` or something)? You then translate these when applying the option...

Reply:
other plugins that uses encoding don't expose levels in their config and use the default one unlike my implementation so that's why I added it here not in internal but we can probably add it in a follow up PR. I have the feeling that merging PRs takes time and this feature will not get released because of this despite that the code is ok.

Reply:
https://github.com/influxdata/telegraf/pull/13424

I made a PR for this here. After the zstd and this PR gets feedback. I'll start working on refactoring this PR.

Reply:
Well we could have done this in parallel by abstracting the actual compression... Anyway, I think we are running out of time for this release window so we will merge this after the release of 1.27.0 and this has to wait for 1.28.0...